lib/picotls.c

```diff
@@ -2424,6 +2439,7 @@ static int client_handle_encrypted_extensions(ptls_t *tls, ptls_iovec_t message,
     static const ptls_raw_extension_t no_unknown_extensions = {UINT16_MAX};
     ptls_raw_extension_t *unknown_extensions = (ptls_raw_extension_t *)&no_unknown_extensions;
     int ret, skip_early_data = 1;
+    const uint8_t *server_offered_cert_type = NULL;
```

Comment:
I think we can do 
```suggestion
    const uint8_t server_offered_cert_type = PTLS_CERTIFICATE_TYPE_X509;`
```
and ...